### PR TITLE
Added basic Pet structure and adoptPet method for basic initialization

### PR DESCRIPTION
**Pull request by jeremydelacruz**
This is pretty basic but open to extension once we figure out how we want to implement ticking down different pet stats (i.e. hunger, happiness, thirst, exercise, etc.). I've implemented some basic validation in `adoptPet()` to check for a valid animal type and non-empty name value, but let me know what else I can add for this story.

### CODE REVIEW

File: src/pages/Game.js [hunk 1]
```diff
@@ -11,38 +11,45 @@ import Animations from '../animations.js'
...
+
+    buildPetDisplay() {
+        const { name, animal } = this.state.pet;
+        return `${name} the ${animal[0].toUpperCase() + animal.substr(1)}`;
     }
 
     render() {
-        let displayName = this.MockPet.name + " the " + this.MockPet.type[0].toUpperCase() + this.MockPet.type.substr(1);
+        if (!this.state.pet) {
+            console.log(this.props);
```
Comment (Bailey-Pearson) [inline src/pages/Game.js]: May want to remove the console.log()

File: src/pages/Game.js [hunk 2]
```diff
@@ -11,38 +11,45 @@ import Animations from '../animations.js'
...
+            return 'Wait... where\'s your pet?! :(';
+        }
 
+        const { pet } = this.state;
+        console.log(pet);
```
Comment (Bailey-Pearson) [inline src/pages/Game.js]: Also this one

File: src/pages/StartingPage.js [hunk 3]
```diff
@@ -6,8 +6,47 @@ import Banner from '../components/Banner.js'
...
+    submitHandler() {
+        const adoptedPet = Pet.adoptPet(this.state.selectedPet, this.state.petName);
+        this.props.history.push({
+            pathname: '/game',
+            state: { pet: adoptedPet },
+        });
+    }
 
     render() {
         return ([
```
Comment (Bailey-Pearson) [inline src/pages/StartingPage.js]: remove the `[` and `]` surrounding the return statement and that last key error will go away

**Review by Bailey-Pearson — CHANGES_REQUESTED** [review]
I think overall the changes look good and work correctly. I do think you should throw in some tests to ensure that the name and pet are being pushed and so we can say we are following our process.

File: src/pages/StartingPage.js (hunk 3 above)
Comment (jeremydelacruz) [reply src/pages/StartingPage.js]: Yeah this was here before, was unsure if it had a functional reason, but didn't try that

**Review by Bailey-Pearson — APPROVED** [review]
Looks great and you really went all in using jest with your tests

**Comment by jeremydelacruz** [comment]
go big or go home

**Pull request merged**